src/measure_theory/function/convergence_in_measure.lean

```diff
@@ -0,0 +1,367 @@
+        (ennreal.of_real_coe_nnreal).le },
+    have : ∀ k, s ⊆ ⋃ i (hik : k ≤ i), S i := λ k, infi_le (λ k, ⋃ i (hik : k ≤ i), S i) k,
+    refine λ k, (measure_mono (this k)).trans ((measure_Union_le _).trans _),
+    have hμ_if_eq : ∀ i, μ (⋃ (hik : k ≤ i), S i) = if k ≤ i then μ (S i) else 0,
+    { intro i, split_ifs; simp only [h, measure_empty, set.Union_true, set.Union_false] },
+    rw tsum_congr hμ_if_eq,
+    have tsum_le_tsum : ∑' i, ite (k ≤ i) (μ (S i)) 0 ≤ ∑' i, ite (k ≤ i) (2⁻¹ ^ i) 0,
+    { refine tsum_le_tsum (λ i, _) ennreal.summable ennreal.summable,
+      split_ifs; simp only [hμS_le i, nonpos_iff_eq_zero] },
+    refine tsum_le_tsum.trans _,
+    suffices tsum_eq_of_real_tsum : ∑' i, ite (k ≤ i) ((2 : ℝ≥0∞)⁻¹ ^ i) 0
+      = ennreal.of_real (∑' i, ite (k ≤ i) (2⁻¹ ^ i) 0),
+    { rw tsum_eq_of_real_tsum,
+      exact ennreal.of_real_le_of_real (tsum_geometric_inv_two_ge k).le },
+    rw ennreal.of_real_tsum_of_nonneg,
+    { refine tsum_congr (λ i, _),
+      split_ifs,
+      { rw [ennreal.of_real_pow (inv_nonneg.mpr zero_le_two) i,
+        ← ennreal.of_real_inv_of_pos zero_lt_two, ennreal.of_real_bit0 zero_le_one,
+        ennreal.of_real_one] },
+      { exact ennreal.of_real_zero.symm } },
+    { intro n,
+      split_ifs,
+      { refine pow_nonneg _ _, norm_num },
+      { exact le_rfl } },
+    { refine summable.summable_of_eq_zero_or_self summable_geometric_two (λ i, _),
+      simp only [one_div, inv_eq_zero, not_le, inv_pow₀, zero_eq_inv],
+      exact (ite_eq_or_eq _ _ _).symm, }, },
+  have h_tendsto : ∀ x ∈ sᶜ, tendsto (λ i, f (ns i) x) at_top (𝓝 (g x)),
+  { refine λ x hx, metric.tendsto_at_top.mpr (λ ε hε, _),
+    simp_rw [s, set.compl_Inter, set.compl_Union, set.mem_Union, set.mem_Inter] at hx,
+    obtain ⟨N, hNx⟩ := hx,
+    obtain ⟨k, hk_lt_ε⟩ := h_lt_ε_real ε hε,
+    refine ⟨max N (k - 1), λ n hn_ge, lt_of_le_of_lt _ hk_lt_ε⟩,
+    specialize hNx n ((le_max_left _ _).trans hn_ge),
+    have h_inv_n_le_k : (2 : ℝ)⁻¹ ^ n ≤ 2⁻¹ ^ ((k : ℝ) - 1),
+    { rw [← real.rpow_nat_cast],
+      refine real.rpow_le_rpow_of_exponent_ge ((one_div (2 : ℝ)) ▸ one_half_pos)
+        (inv_le_one one_le_two) _,
+      rw [sub_le_iff_le_add, ← nat.cast_add_one, nat.cast_le],
+      exact (le_tsub_add.trans (add_le_add_right (le_max_right _ _) 1)).trans
+        (add_le_add_right hn_ge 1) },
+    refine le_trans _ h_inv_n_le_k,
+    rw [set.mem_compl_iff, set.nmem_set_of_eq, not_le] at hNx,
+    exact hNx.le },
+  rw ae_iff,
+  refine ⟨exists_seq_tendsto_ae.seq_tendsto_ae_seq_strict_mono hfg,
+    measure_mono_null (λ x, _) hμs⟩,
+  rw [set.mem_set_of_eq, ← @not_not (x ∈ s), not_imp_not],
+  exact h_tendsto x,
+end
+
+end
+
+section tendsto_in_measure_of
+
+variables [measurable_space E] [normed_group E] [borel_space E] [has_measurable_sub₂ E] {p : ℝ≥0∞}
+variables {f : ℕ → α → E} {g : α → E}
+
+private lemma tendsto_in_measure_of_tendsto_snorm_of_measurable
```

Comment:
We try to avoid private lemmas in mathlib. You can have it as a normal lemma, and add a docstring saying that this is superceded by ...